### ISSUE TITLE
Incorrect jacfwd result for logsumexp with zero b coefficient

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#37633]
## Description
`jax.scipy.special.logsumexp` gives an incorrect forward-mode Jacobian when the `b` argument contains an exact zero.

In the example below, `logsumexp(jnp.zeros(3), b=x)` is mathematically equivalent to `log(jnp.sum(x))`, because `exp(0) = 1`. Therefore, both functions should have the same Jacobian.

However, `jax.jacfwd` returns a zero derivative for the entry where `b == 0`, while the manually equivalent expression returns the expected nonzero derivative.

## Minimal reproducible example:

```
import jax
import jax.numpy as jnp

fun_lse = lambda x: jax.scipy.special.logsumexp(jnp.zeros(3), axis=0, b=x)
fun_man = lambda x: jnp.log(jnp.sum(x))

a = jnp.array([2.0, 0.0, 5.0])

print(f"jacfwd(logsumexp) at {a.tolist()}: {jax.jacfwd(fun_lse)(a)}")
print(f"jacfwd(manual)    at {a.tolist()}: {jax.jacfwd(fun_man)(a)}")
```

## Actual result:

```
An NVIDIA GPU may be present on this machine, but a CUDA-enabled jaxlib is not installed. Falling back to cpu.
jacfwd(logsumexp) at [2.0, 0.0, 5.0]: [0.14285715 0.         0.14285715]
jacfwd(manual)    at [2.0, 0.0, 5.0]: [0.14285715 0.14285715 0.14285715]
```

## Expected result:

```
jacfwd(logsumexp) should match jacfwd(manual):

[0.14285715 0.14285715 0.14285715]
```

### System info (python version, jaxlib version, accelerator, etc.)

```
WARNING:2026-05-07 00:33:38,915:jax._src.xla_bridge:794: An NVIDIA GPU may be present on this machine, but a CUDA-enabled jaxlib is not installed. Falling back to cpu.
jax:    0.6.2
jaxlib: 0.6.2
numpy:  2.2.6
python: 3.10.20 (main, Mar 11 2026, 17:46:40) [GCC 14.3.0]
device info: cpu-1, 1 local devices
process_count: 1
platform: uname_result(system='Linux', node='MSI', release='5.15.167.4-microsoft-standard-WSL2', version='#1 SMP Tue Nov 5 00:21:55 UTC 2024', machine='x86_64')

nvidia-smi:
NVIDIA-SMI 545.23.06
Driver Version: 545.92
CUDA Version: 12.3
GPU: NVIDIA GeForce RTX 3080, 8192 MiB

Note: Reproduced on CPU fallback because CUDA-enabled jaxlib is not installed.
```

**Comment by rehan243** [comment]
Oh, interesting — we ran into something kinda similar with `logsumexp` under the hood in a custom loss function. It wasn’t the `b` arg exactly, but the behavior around zeros and how the autodiff handles sparsity was definitely weird.

The thing is, `logsumexp`'s gradient relies on `softmax`, and when `b=0`, you're effectively multiplying by a mask that kills the contribution of that term in the gradient. It’s like the chain rule gets short-circuited because zero times anything is still zero. The manual version sidesteps this because it doesn’t "see" the zero directly during autodiff — the `sum` propagates the sensitivity correctly.

I think this might be a fundamental JAX AD thing with how it treats multiplication by exact zero as non-differentiable. Not sure there's a clean fix without some tweaks to how `logsumexp` handles the `b` parameter internally.

**Comment by jakevdp** [comment]
This issue seems to be a duplicate of #22398. We briefly landed a candidate fix in #22836, and another in https://github.com/jax-ml/jax/commit/f00878a932f8c6e6fa17b8c51c966afa52735d45, but both had to be rolled back because they caused other issues downstream.

I think a custom JVP is probably the right approach here, given the non-autodiffable manipulations that `logsumexp` is fundamentally built on.

I'm going to close this issue and we can continue tracking it in #22398.
